### PR TITLE
Document supported k8s versions for server

### PR DESCRIPTION
**Pull request by ryanwohara**
SERVER-1498

# Description
Recommending k8s versions for our server versions.

# Reasons
https://circleci.atlassian.net/browse/SERVER-1498
We don't yet support k8s 1.22 until the 3.3 release

### CODE REVIEW

**Review by devinmarieb — APPROVED** [review]
Looks good, I'll leave it to @rosieyohannan to merge after her approval.

**Comment by rosieyohannan** [comment]
Thank you 🙏

**Pull request merged**
